core/templates/a_hash_map.h

```diff
@@ -343,6 +343,28 @@ class AHashMap {
 		return nullptr;
 	}
 
+	TValue &get(const TKey &p_key, const TValue &p_default) {
+		uint32_t pos = 0;
+		uint32_t hash_pos = 0;
+		bool exists = _lookup_pos(p_key, pos, hash_pos);
+
+		if (unlikely(!exists)) {
+			return elements[_insert_element(p_key, p_default, _hash(p_key))].value;
+		}
```

Comment:
If you goal is speed, then you should avoid calculating the hash twice. Typically the cost of the hash makes up the bulk of the cost of accessing HashMaps.